### PR TITLE
Order connectorProvider on datasource picker

### PR DESCRIPTION
**Pull request by MaisonnatM**
Add search input and order as follow: connectorProvider > empty > webcrawler

### CODE REVIEW

**Comment by MaisonnatM** [comment]
Capture d’écran 2024-01-22 à 12 37 18

File: front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx [hunk 1]
```diff
@@ -138,19 +139,63 @@ function PickDataSource({
...
+            .connectorProvider !==
+          CONNECTOR_CONFIGURATIONS.webcrawler.connectorProvider
+      )
+      .map(
+        (key) =>
+          CONNECTOR_CONFIGURATIONS[key as keyof typeof CONNECTOR_CONFIGURATIONS]
+            .connectorProvider
+      );
+
+    if (aConnector === CONNECTOR_CONFIGURATIONS.webcrawler.connectorProvider) {
```
Comment (spolu) [inline front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx:162]: Doesn't it place webcrawler websites at the top?

File: front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx [hunk 2]
```diff
@@ -138,19 +139,63 @@ function PickDataSource({
   show: boolean;
   onPick: (dataSource: DataSourceType) => void;
 }) {
+  const [query, setQuery] = useState("");
+
+  // Order in the following format : connectorProvider > empty > webcrawler
+  const orderedDataSources = dataSources.sort((a, b) => {
```
Comment (spolu) [inline front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx:145]: Can you move this function in `lib/assistant.ts` where we already have some assistant / related sorting. Name LGTM :+1:

**Review by spolu — COMMENTED** [review]
LGTM ! Left 2 minor comments to address :+1:

File: front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx (hunk 1 above)
Comment (MaisonnatM) [reply front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx:162]: Nop, when webcrawler is `a` then the `b` element should be upper (so I return 1)

Comment (spolu) [reply front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx:162]: Ok!

**Review by spolu — APPROVED** [review]

**Comment by spolu** [comment]
Will deploy

File: front/lib/assistant.ts [hunk 3]
```diff
@@ -99,24 +101,40 @@ export function compareAgentsForSort(
   return 0; // Default: keep the original order
 }
 
-export function assistantUsageMessage({
```
Comment (spolu) [inline front/lib/assistant.ts:102]: Oopsy you erased this function @MaisonnatM

**Pull request merged**